src/olympia/editors/views.py

```diff
@@ -587,56 +625,24 @@ def review(request, addon):
     try:
         show_diff = version and (
             addon.versions.exclude(id=version.id).filter(
-                files__isnull=False, created__lt=version.created,
+                channel=channel,
+                files__isnull=False,
+                created__lt=version.created,
                 files__status=amo.STATUS_PUBLIC).latest())
     except Version.DoesNotExist:
         show_diff = None
 
     # The actions we should show a minimal form from.
     actions_minimal = [k for (k, a) in actions if not a.get('minimal')]
 
-    versions = (Version.unfiltered.filter(addon=addon)
+    versions = (Version.unfiltered.filter(addon=addon, channel=channel)
                                   .exclude(files__status=amo.STATUS_BETA)
                                   .order_by('-created')
                                   .transform(Version.transformer_activity)
                                   .transform(Version.transformer))
 
-    class PseudoVersion(object):
-        def __init__(self):
-            self.all_activity = []
-
-        all_files = ()
-        approvalnotes = None
-        compatible_apps_ordered = ()
-        releasenotes = None
-        status = 'Deleted'
-        deleted = True
-        channel = amo.RELEASE_CHANNEL_LISTED
-
-        @property
-        def created(self):
-            return self.all_activity[0].created
-
-        @property
-        def version(self):
-            return (self.all_activity[0].activity_log
-                        .details.get('version', '[deleted]'))
-
-    # Grab review history for deleted versions of this add-on
-    # Version are soft-deleted now but we need this for older deletions.
-    comments = (CommentLog.objects
-                .filter(activity_log__action__in=amo.LOG_REVIEW_QUEUE,
-                        activity_log__versionlog=None,
-                        activity_log__addonlog__addon=addon)
-                .order_by('created')
-                .select_related('activity_log'))
-
-    comment_versions = defaultdict(PseudoVersion)
-    for c in comments:
-        c.version = c.activity_log.details.get('version', c.created)
-        comment_versions[c.version].all_activity.append(c)
-
-    all_versions = comment_versions.values()
+    all_versions = (_get_deleted_versions(addon)
+                    if channel == amo.RELEASE_CHANNEL_LISTED else [])
```

Comment:
Add a comment explaining why we only care about listed channel here (kinda redundant with the docstring I'm asking for above but let's be more explicit :)